### PR TITLE
[MIRROR] Ordnance Content Update: Scientific Papers [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/62284
--------------------
## How do I play/test/operate this?
1.  Download NT Frontier on any modular computers. It should debrief you on what experiments are available and how to publish.
2.  If you want to do a bomb experiment, make sure it's captured by the doppler array (as usual) and then print the experiments into a disk and publish it. 
3. If you want to do a gas experiment, make the gas and either pump it into a tank and 1) overpressurize it with a "clear" gas like N2 or 2) overpressurize tanks with the gas itself. Make sure you do the overpressurizing in the compressor machine. When tanks are destroyed/ejected leaked gas will get recorded. Print it into a disk and publish it.
4. For publication, the file needs to be directly present **inside** the computer's HDD. This means you need to copy it first with the file manager.
5. Fill the data (if desired, it will autofill with boiler plate if you dont) and send away!
6. Doing experiments unlock nodes, while doing them well unlocks boosts (which are discounts but slightly more restrictive) which are purchaseable with NT Frontier.
7. If you are testing this and have access to admin tools, there are various premade bombs under `obj/effect/spawner/newbomb`

## About The Pull Request
A doc I wrote detailing the why and what part of this PR.
https://hackmd.io/JOakSYVMSh2zU2YL5ju_-Q
TLDR: Added new experiment to toxins, added the framework for those experiments existing. Experiments comes in gas synthesis and also bombs but with more parameters. Experiments needs to be published through papers, various choices to be made there.

- [x] Combine implosion compressor and bomb processor into one.
- [x] Make the UI up to standard.
- [x] Add a sprite for the compressor.
- [x] Map changes.

<details>
<summary> Codersprites </summary>
They're not very good but they're serviceable at least.
<br/>
<image src="https://user-images.githubusercontent.com/54709710/141190268-f88eba07-a2f7-4c1f-8943-7140424ac948.png">
<br/>
<image src="https://user-images.githubusercontent.com/54709710/141190440-bf26a4da-829d-49f5-a4aa-983910dcff90.png">
<br/>
<image src="https://user-images.githubusercontent.com/54709710/141189632-e9d5f875-7ed4-4948-8f3f-2226abb52d95.png">
<image src="https://user-images.githubusercontent.com/54709710/141189743-f4cdbaf2-d3c9-4b43-9ece-91a1c51e0e60.png">
<image src="https://user-images.githubusercontent.com/54709710/141189892-aa42c875-32db-41b3-9c49-d04f1a26b7b3.png">
</details>

<details>
<summary> Doppler malf ban he </summary>
<br/>
small change but i couldn't resist making the joke
</details>

Implementation notes:
Because of how paper works, ordnance experiments are handled outside of experiment_handler components. My reasoning for this is twofold: 
1) The experiments will be completed manually on publication and if the experiment isn't unlocked yet it will still be completed. 
2) Experiment handler datums have several procs which require an atom-level parent, and I figured this is the most sensible and cleanest way to implement this without changing the experiment handler datum too much. 
I won't mind changing this if requested by a maintainer.

Small change to `/obj/machinery/proc/power_change()` signal ordering to adjust the state first and then send the signal. Didn't found any other usage of this signal except mine but barge down my door if it broke something.

Rewrote the ttv merge_gases() code to be slightly more readable.
A small code improvement for thermomachine to use tofixed (my fault).

## Why It's Good For The Game
Adds content to ordnance, more experiments, more variety. 
More sensible learning curve, better player feedback.
Pretty much refer to my hackmd for a more substantial justification.

## Changelog
:cl:
add: Ordnance have been updated to enable the publication of papers
add: Several new explosive and gas synthesis experiments have been added to ordnance
add: Anomaly compressor has been TGUIzed and now supports simulating the reaction of the gases inside the ttv.
add: New tank compressor machine for toxins. You can overpressurize tanks with exotic gases and complete experiments.
balance: Several techweb nodes are locked and require toxin experiments to complete.
balance: Toxins can purchase boosts for various techweb nodes.
balance: You no longer need to anchor doppler arrays for it to work.
qol: Doppler array and implosion compressor now supports deconstruction, implosion compressor construction added.
qol: Doppler now emits a red light to denote it's direction and it being on. Doppler not malf.
code: Implosion compressor renamed to anomaly refinery.
code: Created a new program tab "Science" for the downloader app. Removed Robotics.
refactor: Reworked the code for bombspawner (used in the cuban pete arcade game)
/:cl:

## UI:
<details>
<summary>Tank Compressor</summary>

</details>

<details>
<summary>Publisher</summary>

</details>

<details>
<summary>Doppler</summary>

</details>

<details>
<summary>Detonation Chamber / Anomaly Compressor</summary>

</details>

### CODE REVIEW

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 1]
```diff
@@ -123,19 +131,26 @@
...
+<<<<<<< HEAD
 	location.PolluteListTurf(list(/datum/pollutant/smoke = 15, /datum/pollutant/carbon_air_pollution = 5), POLLUTION_ACTIVE_EMITTER_CAP) //SKYRAT EDIT ADDITION
+=======
+	if(reference)
+		volume = 0
+		var/list/cached_results = reference.reaction_results
+		for (var/reaction in SSair.hotspot_reactions)
+			volume += cached_results[reaction] * FIRE_GROWTH_RATE
+		temperature = reference.temperature
+>>>>>>> eeb54659314 (Ordnance Content Update: Scientific Papers (#62284))
```
Comment (GoldenAlpharex) [inline code/modules/atmospherics/environmental/LINDA_fire.dm]:
```suggestion
	location.PolluteListTurf(list(/datum/pollutant/smoke = 15, /datum/pollutant/carbon_air_pollution = 5), POLLUTION_ACTIVE_EMITTER_CAP) //SKYRAT EDIT ADDITION
	if(reference)
		volume = 0
		var/list/cached_results = reference.reaction_results
		for (var/reaction in SSair.hotspot_reactions)
			volume += cached_results[reaction] * FIRE_GROWTH_RATE
		temperature = reference.temperature
```

**Review by GoldenAlpharex — APPROVED** [review]
Almost good to go, now to update the maps.

**Review by GoldenAlpharex — APPROVED** [review]

File: code/modules/clothing/suits/jobs.dm [hunk 2]
```diff
@@ -72,7 +72,7 @@
 	name = "noir trenchcoat"
 	desc = "A hard-boiled private investigator's dark trenchcoat."
 	icon_state = "noirdet"
-	inhand_icon_state = "greydet"
+	inhand_icon_state = "noirdet"
```
Comment (GoldenAlpharex) [inline code/modules/clothing/suits/jobs.dm]: I am
```suggestion
	inhand_icon_state = "greydet"
```

File: code/modules/clothing/under/jobs/security.dm [hunk 3]
```diff
@@ -113,14 +113,14 @@
 	name = "noir suit"
 	desc = "A hard-boiled private investigator's dark suit, complete with tie clip."
 	icon_state = "noirdet"
-	inhand_icon_state = "greydet"
+	inhand_icon_state = "noirdet"
```
Comment (GoldenAlpharex) [inline code/modules/clothing/under/jobs/security.dm]: An
```suggestion
	inhand_icon_state = "greydet"
```

File: code/modules/clothing/under/jobs/security.dm [hunk 4]
```diff
@@ -113,14 +113,14 @@
...
 	alt_covers_chest = TRUE
 
 /obj/item/clothing/under/rank/security/detective/noir/skirt
 	name = "noir suitskirt"
 	desc = "A hard-boiled private investigator's grey suitskirt, complete with tie clip."
 	icon_state = "noirdet_skirt"
-	inhand_icon_state = "greydet"
+	inhand_icon_state = "noirdet"
```
Comment (GoldenAlpharex) [inline code/modules/clothing/under/jobs/security.dm]: idiot
```suggestion
	inhand_icon_state = "greydet"
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Review by GoldenAlpharex — APPROVED** [review]
